### PR TITLE
Output rendering application in meta tag

### PR DESCRIPTION
**Pull request by edds**
I am interested in thoughts on this being a thing that should exist as much as the technical implementation of this change. Do you think this is a good idea?

----

It would be really useful to be able to segment analytics by rendering
application. It would let us have things like JavaScript errors per
application and page timings per application.

This change takes the class name of the currently running application
and adds it as a meta tag which can then be read by JavaScript.

### CODE REVIEW

**Comment by rboulton** [comment]
I think this would be incredibly useful, and am +:100: on it being done, for exactly the reasons you outline.

**Comment by dsingleton** [comment]
This is a great idea. Being able to segment errors, timings &c by app is super useful.

File: lib/slimmer/app.rb [hunk 1]
```diff
@@ -110,5 +110,9 @@ def rewrite_response(env, response)
     def strip_slimmer_headers(headers)
       headers.reject {|k, v| k =~ /\A#{Headers::HEADER_PREFIX}/ }
     end
+
+    def application_class_name
+      @app.class.parent_name if @app.class.respond_to?(:parent_name)
```
Comment (dsingleton) [inline lib/slimmer/app.rb]: It's a little vague to me what sort of return value we'd expect here, i assume something like `FinderFrontend`?

**Comment by fofr** [comment]
:heart: :+1:

**Comment by bradleywright** [comment]
:+1:

**Comment by bradleywright** [comment]
Is it worth making this configurable, in case the autodiscovery doesn't work as intended?

**Comment by edds** [comment]
Ugh. Trying this on some more applications if other middleware is loaded it doesn't do what I was expecting. Going to have to try another approach.

**Comment by edds** [comment]
I have tried a subtly different approach as looking at the `@app` in middleware just gives you the next item in the stack rather than the actual application.

Reopend over here https://github.com/alphagov/slimmer/pull/126 because you can't reopen a closed pull request with a force push.

**Pull request closed without merging**